tests/unit/test_charm.py

```diff
@@ -12,24 +12,32 @@
 import pytest
 from ops.testing import Harness
 
+import synapse
+from charm import SynapseCharm
 from constants import (
     SYNAPSE_COMMAND_PATH,
+    SYNAPSE_CONFIG_PATH,
     SYNAPSE_CONTAINER_NAME,
     SYNAPSE_NGINX_CONTAINER_NAME,
     SYNAPSE_NGINX_PORT,
     SYNAPSE_SERVICE_NAME,
     TEST_SERVER_NAME,
+    TEST_SERVER_NAME_CHANGED,
 )
 from pebble import PebbleServiceError
 
 
-def test_synapse_pebble_layer(harness_server_name_configured: Harness) -> None:
+def test_synapse_pebble_layer(harness: Harness, monkeypatch: pytest.MonkeyPatch) -> None:
     """
     arrange: charm deployed.
     act: start the Synapse charm, set Synapse container to be ready and set server_name.
     assert: Synapse charm should submit the correct Synapse pebble layer to pebble.
     """
-    harness = harness_server_name_configured
+    monkeypatch.setattr(synapse, "get_version", lambda *_args, **_kwargs: "")
+    harness.update_config({"server_name": TEST_SERVER_NAME})
```

Comment:
Is this executed for every tests and, if so, have you considered moving it to the fixture?

Reply:
Good point, I was hesitating between this and moving it to a fixture because keeping it here I think gives a better idea of what we are doing in the tests. But I'll move it to a fixture.

Reply:
I have made the changes here : https://github.com/canonical/synapse-operator/pull/36/commits/c331ecb1a5a485378059e08ebc76026e9c7e7c78

Reply:
Maybe Arturo was referring to the configuration of setting the server name? That could be a fixture as well since it seems to be done a lot

Reply:
I might have missed that. I have also move it to a fixture and did an overall cleanup of those duplicates